### PR TITLE
Validating Sampler

### PR DESCRIPTION
**Pull request by dstrain115**
- Wrapper around sampler to do device related validation.
- This will be used in AbstractEngine in order to centralize
and simplify validation.

### CODE REVIEW

File: cirq-google/cirq_google/engine/validating_sampler.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+           checks.  Note that this function takes a list of circuits and
+           sweeps so that batch functionality can also be tested.
+    """
+
+    def __init__(
+        self,
+        *,
+        device: Optional[cirq.Device] = None,
+        validator: Optional[VALIDATOR_TYPE] = None,
+        sampler: cirq.Sampler = cirq.Simulator(),
```
Comment (MichaelBroughton) [inline cirq-google/cirq_google/engine/validating_sampler.py:42]: Nit: Can you move this docstring to the `__init__` function, put the arguments in the same order and be sure to surround things like \`cirq.Device\` in backticks for linking to take place on the devsite ?

File: cirq-google/cirq_google/engine/validating_sampler.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    """Wrapper around cirq.Sampler that performs device validation.
+
+    This sampler will delegate to the wrapping sampler after
+    performing validation on the circuit(s) given to the sampler.
+
+    Args:
+        sampler: sampler wrapped by this object.  After validating,
+            samples will be returned by this enclosed cirq.Sampler.
+        device: cirq.Device that will validate_circuit before sampling.
+        validator: A callable that will do any additional validation
```
Comment (MichaelBroughton) [inline cirq-google/cirq_google/engine/validating_sampler.py:31]: Should it also take the repetition numbers as well ?

**Review by MichaelBroughton — APPROVED** [review]

File: cirq-google/cirq_google/engine/validating_sampler.py (hunk 1 above)
Comment (dstrain115) [reply cirq-google/cirq_google/engine/validating_sampler.py]: Done.

File: cirq-google/cirq_google/engine/validating_sampler.py (hunk 2 above)
Comment (dstrain115) [reply cirq-google/cirq_google/engine/validating_sampler.py]: Good idea.  Done.

**Pull request merged**